app/response_processor.py

```diff
@@ -106,33 +106,41 @@ def send_receipt(self, decrypted_json):
             receipt_json = {
                 'tx_id': decrypted_json['tx_id'],
                 'collection': {
-                    'exercise_sid': decrypted_json['collection']['exercise_sid']
+                    'exercise_sid':
+                    decrypted_json['collection']['exercise_sid']
                 },
                 'metadata': {
                     'ru_ref': decrypted_json['metadata']['ru_ref'],
                     'user_id': decrypted_json['metadata']['user_id']
                 }
             }
         except KeyError as e:
-            self.logger.error("Unsuccesful publish, missing key values", error=e)
+            self.logger.error(
+                "Unsuccesful publish, missing key values", error=e)
             raise QuarantinableError
 
+        try:
+            case_id = decrypted_json['case_id']
+            receipt_json['case_id'] = case_id
+        except KeyError:
+            # Don't do anything, as this survey originated from RRM
+            pass
```

Comment:
it snever a good practice to leave erro with no trace , I woudl log a warning or a debug in case.

Reply:
Any log for this?